LandSandBoat/server
# [core] Audit battleutils::CalculateSpellRecastTime, change Haste spells to wind element

**_I affirm:_**

## What does this pull request do?

* add Inspiration FC bonus
* adjust FC cap to 40% recast cap like retail
* Adjust light/dark arts logic
* Only go over 80% spell reduction cap with SCH AF3 boots bonus with correct weather
* Standardize percent based reduction calculations

## Steps to test these changes

Verify Alacrity/Celerity + weather with AF3 SCH boots reduces spell recast to 90% of normal when fully hasted out
Verify recast reduction with Inspiration merits + Vallation/Valiance from a RUN
Verify 10% reduction from light/dark arts alone
